include/lsst/meas/base/PixelFlags.h

```diff
@@ -94,10 +110,12 @@ class PixelFlagsAlgorithm : public SimpleAlgorithm {
     ) const;
 
 private:
-
+    template <typename MaskedImageT>
+    class FootprintBits;
+    typedef afw::image::MaskedImage<float> MaskedImageT;
+    void updateFlags(const KeyMap *,const FootprintBits<MaskedImageT> *,afw::table::SourceRecord & ) const;
```

Comment:
I don't think that there's any question that const reference should be used wherever appropriate (but e.g. iterators are usually passed by value, not const reference). My comment was only about *non* const refs.